core/thread/id.go

```diff
@@ -294,6 +294,19 @@ func (i ID) Loggable() map[string]interface{} {
 	}
 }
 
+func (i ID) MarshalJSON() ([]byte, error) {
+	return i.Bytes(), nil
```

Comment:
@sanderpick I know you mentioned somehow using the multibase methods in here, but I thought I'd try to just use the existing methods that convert `ID` to and from `[]byte`. When having `encoding/json` marshal a `Person` in the test below, I get this error:
```json: error calling MarshalJSON for type thread.ID: invalid character '\x01' looking for beginning of value```
Ideas?

Reply:
To clarify, `json/encoding` chokes on the bytes returned from here and gives that error.

Reply:
Nice, good call on using existing methods. Not sure what's up with that. Taking a look.

Reply:
Looking here @asutula - Do you know what the original error was that led to this work? Running your new test without the custom marshalers seems to be fine. It looks like the `json` package falls back to using `TextMarshaler`, which is already implemented on `thread.ID`. I copied that from the `CID` implementation... maybe they included that to support JSON.

I changed the test to:
```
func TestJSON(t *testing.T) {
	i := NewIDV1(Raw, 32)
	t.Logf("New ID: %s", i.String())

	data, err := json.Marshal(i)
	if err != nil {
		t.Errorf("failed to marshal ID %s: %s", i.String(), err)
	}
	t.Logf("JSON Encoded ID: %s", string(data))

	i2 := ID{}
	err = json.Unmarshal(data, &i2)
	if err != nil {
		t.Errorf("failed to unmarshal ID: %s", err)
	}

	if i.str != i2.str {
		t.Errorf("ID string %s != %s", i.str, i2.str)
	}
}
```

Which just uses the `json` package directly.

Reply:
Ah yea, brain is foggy today it seems. I think this has diverged from the original problem. I'll push a test that illustrates the original problem which has to do with json schemas that include thread ID.